### PR TITLE
feat(typescript-angular): add support for Angular V17

### LINKED ISSUE #17301: [REQ] Support of Angular 17 in generator typescript-angular
### Is your feature request related to a problem? Please describe.

Some dependencies in the generated package by typescript-angular generator are not compatible with the dependencies of Angular 17.

## Describe the solution you'd like

When angular version is set to 17 or higher, use the following dependencies:

```
      "rxjs": "~7.8.0",
      "typescript": ">=4.9.3 <5.3.0",
      "zone.js": "~0.14.2"
```

## Describe alternatives you've considered

For the moment I override the dependencies in the package.json of my application in a ovverides section such as:
```
  "overrides": {
    "@myorg/my-api-client": {
      "@angular/common": "^17.0.5",
      "@angular/compiler": "^17.0.5",
      "@angular/compiler-cli": "^17.0.5",
      "@angular/core": "^17.0.5",
      "@angular/platform-browser": "^17.0.5",
      "rxjs": "^7.8.0",
      "typescript": ">=4.9.3 <5.3.0",
      "zone.js": "^0.14.2"
    },
```

### PR DESCRIPTION
**Pull request by alethyst**
### Changes

- Add support for Angular V17
- Add support for `transferCache` property in request options

### Relevant technical committee members: 
@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03) @mkusaka (2022/04)

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by jerometambo** [comment]
yes please please please :D

**Comment by alethyst** [comment]
Hello @wing328, is it still possible to get this merged in before cutting the release for 7.3.0?  It'll be helpful for me to know so I can plan around it. Thank you.

**Comment by saadsme** [comment]
Need this added asap!!!

File: modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache [hunk 1]
```diff
@@ -283,6 +283,13 @@ export class {{classname}} {
             localVarHttpContext = new HttpContext();
         }
 {{/httpContextInOptions}}
+{{#httpTransferCacheInOptions}}
+
+        let localVarTransferCache: boolean | undefined = options && options.transferCache;
+        if (localVarTransferCache === undefined) {
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache:289]: how about using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing?

Comment (macjohnny) [reply modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache:289]: (could be done in a separate PR, in order to unblock this one)

**Review by macjohnny — APPROVED** [review]
Thanks for your contribution!

File: modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache (hunk 1 above)
Comment (alethyst) [reply modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache:289]: I was trying to stay close to the already existing code style that's why it's a little verbose, but sure I'll open another PR for it when I can.

**Comment by alethyst** [comment]
Thanks for merging!

**Comment by thibaudsowa** [comment]
🎉 @macjohnny Do you have an estimation of the next release date?

**Comment by fabsi96** [comment]
> 🎉 @macjohnny Do you have an estimation of the next release date?

Would be also my question 🤙

**Comment by wing328** [comment]
we will cut a release next week (tentatively on monday)

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/milestones has more info

**Pull request merged**